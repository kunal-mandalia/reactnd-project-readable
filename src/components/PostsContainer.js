import React, { Component } from 'react'
import { connect } from 'react-redux'
import Post from './Post'
import Comment from './Comment'
import NewComment from './NewComment'
import { sortBy } from '../helper/functions'
import {
  editPost,
  editComment,
  beginEditPost,
  endEditPost,
  beginEditComment,
  endEditComment,
  votePost,
  voteComment,
  deletePost,
  deleteComment,
  newCommentShow,
} from '../actions/index.js'
import NotFound from './NotFound'

import '../styles/PostsContainer.css'

export class PostsContainer extends Component {

  editPost = (id, newTitle, newBody) => { this.props.editPost(id, newTitle, newBody) }
  deletePost = (id) => { this.props.deletePost(id) }
  beginEditPost = (id) => { this.props.beginEditPost(id) }
  endEditPost = (id) => { this.props.endEditPost(id) }
  beginEditComment = (id) => { this.props.beginEditComment(id) }
  endEditComment = (id) => { this.props.endEditComment(id) }
  editComment = (id, newBody) => { this.props.editComment(id, newBody) }
  deleteComment = (id) => { this.props.deleteComment(id) }
  votePost = (id, upVote) => { this.props.votePost(id, upVote) }
  voteComment = (id, upVote) => { this.props.voteComment(id, upVote) }
  newCommentShow = (parentId) => { this.props.newCommentShow(parentId) }

  render () {
    const { posts, comments, sort, user, filterBy } = this.props
    return (
      <div className='posts-container'>
        {posts.length > 0 ? (
          posts.map(p => (
          <div key={p.id}>
            <Post
              {...p}
              update={this.props.updates[p.id]}
              onEdit={this.beginEditPost}
              onCancel={this.endEditPost}
              onSave={this.editPost}
              onDelete={this.deletePost}
              onVoteUp={() => { this.votePost(p.id, true) }}
              onVoteDown={() => { this.votePost(p.id, false) }}
              onReply={() => { this.newCommentShow(p.id) }}
              isAuthor={user === p.author}
            />
            <div className='posts-comments'>
              {comments.filter(c => c.parentId === p.id )
                .map(c => (
                  <Comment
                    key={c.id}
                    {...c}
                    onEdit={this.beginEditComment}
                    onCancel={this.endEditComment}
                    onSave={this.editComment}
                    onDelete={this.deleteComment}
                    onVoteUp={() => { this.voteComment(c.id, true) }}
                    onVoteDown={() => { this.voteComment(c.id, false) }}
                    update={this.props.updates[c.id]}
                    onReply={() => { this.newCommentShow(p.id) }}
                    isAuthor={user === c.author}
                  />
              ))}
              <NewComment parentId={p.id} />
            </div>            
          </div>
        ))) : (
          <NotFound message='No posts found' filterBy={filterBy} />
        )}
      </div>
    )
  }
}

const mapStateToProps = state => ({})

const mapDispatchToProps = dispatch => ({
  editPost: (id, newTitle, newBody) => dispatch(editPost({ id, title: newTitle, body: newBody })),
  deletePost: (id) => dispatch(deletePost(id)),
  beginEditPost: (id) => dispatch(beginEditPost(id)),
  endEditPost: (id) => dispatch(endEditPost(id)),
  beginEditComment: (id) => dispatch(beginEditComment(id)),
  endEditComment: (id) => dispatch(endEditComment(id)),
  editComment: (id, newBody) => dispatch(editComment({ id, body: newBody })),
  deleteComment: (id) => dispatch(deleteComment(id)),
  votePost: (id, upVote) => dispatch(votePost({ id, upVote })),
  voteComment: (id, upVote) => dispatch(voteComment({ id, upVote })),
  newCommentShow: (parentId) => dispatch(newCommentShow(parentId))
})

export default connect(mapStateToProps, mapDispatchToProps)(PostsContainer)
