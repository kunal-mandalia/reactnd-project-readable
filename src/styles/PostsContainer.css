.posts-container {
  margin-top: 14px;
  margin: 15px auto;
  background-color: #ececec;
  max-width: 600px;
  padding-bottom: 80px;
}

.posts-comments {
  border-left: solid 2px #c2c2c2;
  margin-left: 40px;
}