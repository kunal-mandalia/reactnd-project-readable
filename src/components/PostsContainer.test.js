import React from 'react'
import { PostsContainer } from './PostsContainer'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

const posts = {
  '8xf0y6ziyjabvozdd253nd': {
    id: '8xf0y6ziyjabvozdd253nd',
    timestamp: 1467166872634,
    title: 'Udacity is the best place to learn React',
    body: 'Everyone says so after all.',
    author: 'thingtwo',
    category: 'react',
    voteScore: 6,
    deleted: false 
  },
  '6ni6ok3ym7mf1p33lnez': {
    id: '6ni6ok3ym7mf1p33lnez',
    timestamp: 1468479767190,
    title: 'Learn Redux in 10 minutes!',
    body: 'Just kidding. It takes more than 10 minutes to learn technology.',
    author: 'thingone',
    category: 'redux',
    voteScore: -5,
    deleted: false
  }
}

const comments = {
  '894tuq4ut84ut8v4t8wun89g': { 
    id: '894tuq4ut84ut8v4t8wun89g',
    parentId: "8xf0y6ziyjabvozdd253nd",
    timestamp: 1468166872634,
    body: 'Hi there! I am a COMMENT.',
    author: 'thingtwo',
    voteScore: 6,
    deleted: false,
    parentDeleted: false 
  },
  '8tu4bsun805n8un48ve89': {
    id: '8tu4bsun805n8un48ve89',
    parentId: "8xf0y6ziyjabvozdd253nd",
    timestamp: 1469479767190,
    body: 'Comments. Are. Cool.',
    author: 'thingone',
    voteScore: -5,
    deleted: false,
    parentDeleted: false
  }
}

const mockFn = jest.fn()
const props = {
  sort: { by: 'votes', descending: true },
  editPost: mockFn,
  deletePost: mockFn,
  beginEditPost: mockFn,
  beginEditComment: mockFn,
  endEditComment: mockFn,
  editComment: mockFn,
  deleteComment: mockFn,
  votePost: mockFn,
  voteComment: mockFn,
  newCommentShow: mockFn,
}

const updates = {}

describe(`PostsContainer`, () => {
  const wrapper = shallow(<PostsContainer posts={posts} comments={comments} updates={updates} {...props} />)
  it(`should render ${Object.keys(posts).length} post(s)`, () => {
    expect(wrapper.find('Post')).toHaveLength(Object.keys(posts).length)
  })

  it(`should render ${Object.keys(comments).length} comment(s)`, () => {
    expect(wrapper.find('Comment')).toHaveLength(Object.keys(comments).length)
  })
})
