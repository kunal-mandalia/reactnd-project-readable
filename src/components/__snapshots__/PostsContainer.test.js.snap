// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsContainer should match snapshot 1`] = `
<div
  className="posts-container"
>
  <div>
    <div
      className="post-wrapper"
    >
      <div
        className="post"
      >
        <div
          className="post-row"
        >
          <div
            className="rating"
          >
            <div
              id="upvote-button"
              onClick={[Function]}
            >
              <svg
                className="vote"
                fill="currentColor"
                height={14}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": "grey",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width={14}
              >
                <g>
                  <path
                    d="m37.6 29.7l-3.7 3.7q-0.5 0.4-1 0.4t-1-0.4l-11.9-11.9-11.8 11.9q-0.5 0.4-1.1 0.4t-1-0.4l-3.7-3.7q-0.4-0.4-0.4-1t0.4-1l16.6-16.6q0.4-0.4 1-0.4t1 0.4l16.6 16.6q0.4 0.4 0.4 1t-0.4 1z"
                  />
                </g>
              </svg>
            </div>
            <div
              className="rating-value"
              id="rating-value"
            >
              6
            </div>
            <div
              id="downvote-button"
              onClick={[Function]}
            >
              <svg
                className="vote"
                fill="currentColor"
                height={14}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": "grey",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width={14}
              >
                <g>
                  <path
                    d="m37.6 18l-16.6 16.6q-0.4 0.4-1 0.4t-1-0.4l-16.6-16.6q-0.4-0.4-0.4-1t0.4-1l3.7-3.7q0.5-0.4 1-0.4t1 0.4l11.9 11.9 11.9-11.9q0.4-0.4 1-0.4t1 0.4l3.7 3.7q0.4 0.4 0.4 1t-0.4 1z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <div
            className="post-content"
          >
            <div
              className="inline-edit-read-only inline-edit post-title edit-input-unchanged"
            >
              Udacity is the best place to learn React
            </div>
            <div
              className="inline-edit-read-only inline-edit post-body edit-input-unchanged"
            >
              Everyone says so after all.
            </div>
            <div
              className="action-bar"
            >
              <hr
                className="action-bar-separator"
              />
              <span
                className="action-bar-date"
              >
                 
                &gt;1y
                 
              </span>
              <span
                className="action-bar-author"
              >
                 by 
                thingtwo
                 
              </span>
              <span
                className="action-bar-category"
              >
                 in 
                react
                 
              </span>
              <span
                className="action-bar-actions"
              >
                [
                <a
                  className="action-bar-action"
                  onClick={[Function]}
                >
                   edit 
                </a>
                ]
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="posts-comments"
    >
      <div
        className="comment-wrapper"
      >
        <div
          className="comment"
        >
          <div
            className="comment-row"
          >
            <div
              className="rating"
            >
              <div
                id="upvote-button"
                onClick={[Function]}
              >
                <svg
                  className="vote"
                  fill="currentColor"
                  height={14}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "color": "grey",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 40 40"
                  width={14}
                >
                  <g>
                    <path
                      d="m37.6 29.7l-3.7 3.7q-0.5 0.4-1 0.4t-1-0.4l-11.9-11.9-11.8 11.9q-0.5 0.4-1.1 0.4t-1-0.4l-3.7-3.7q-0.4-0.4-0.4-1t0.4-1l16.6-16.6q0.4-0.4 1-0.4t1 0.4l16.6 16.6q0.4 0.4 0.4 1t-0.4 1z"
                    />
                  </g>
                </svg>
              </div>
              <div
                className="rating-value"
                id="rating-value"
              >
                6
              </div>
              <div
                id="downvote-button"
                onClick={[Function]}
              >
                <svg
                  className="vote"
                  fill="currentColor"
                  height={14}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "color": "grey",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 40 40"
                  width={14}
                >
                  <g>
                    <path
                      d="m37.6 18l-16.6 16.6q-0.4 0.4-1 0.4t-1-0.4l-16.6-16.6q-0.4-0.4-0.4-1t0.4-1l3.7-3.7q0.5-0.4 1-0.4t1 0.4l11.9 11.9 11.9-11.9q0.4-0.4 1-0.4t1 0.4l3.7 3.7q0.4 0.4 0.4 1t-0.4 1z"
                    />
                  </g>
                </svg>
              </div>
            </div>
            <div
              className="comment-content"
            >
              <div
                className="inline-edit-read-only inline-edit comment-body edit-input-unchanged"
              >
                Hi there! I am a COMMENT.
              </div>
              <div
                className="action-bar"
              >
                <hr
                  className="action-bar-separator"
                />
                <span
                  className="action-bar-date"
                >
                   
                  &gt;1y
                   
                </span>
                <span
                  className="action-bar-author"
                >
                   by 
                  thingtwo
                   
                </span>
                <span
                  className="hidden"
                >
                   in 
                   
                </span>
                <span
                  className="action-bar-actions"
                >
                  [
                  <a
                    className="action-bar-action"
                    onClick={[Function]}
                  >
                     edit 
                  </a>
                  ]
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="comment-wrapper"
      >
        <div
          className="comment"
        >
          <div
            className="comment-row"
          >
            <div
              className="rating"
            >
              <div
                id="upvote-button"
                onClick={[Function]}
              >
                <svg
                  className="vote"
                  fill="currentColor"
                  height={14}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "color": "grey",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 40 40"
                  width={14}
                >
                  <g>
                    <path
                      d="m37.6 29.7l-3.7 3.7q-0.5 0.4-1 0.4t-1-0.4l-11.9-11.9-11.8 11.9q-0.5 0.4-1.1 0.4t-1-0.4l-3.7-3.7q-0.4-0.4-0.4-1t0.4-1l16.6-16.6q0.4-0.4 1-0.4t1 0.4l16.6 16.6q0.4 0.4 0.4 1t-0.4 1z"
                    />
                  </g>
                </svg>
              </div>
              <div
                className="rating-value"
                id="rating-value"
              >
                -5
              </div>
              <div
                id="downvote-button"
                onClick={[Function]}
              >
                <svg
                  className="vote"
                  fill="currentColor"
                  height={14}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "color": "grey",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 40 40"
                  width={14}
                >
                  <g>
                    <path
                      d="m37.6 18l-16.6 16.6q-0.4 0.4-1 0.4t-1-0.4l-16.6-16.6q-0.4-0.4-0.4-1t0.4-1l3.7-3.7q0.5-0.4 1-0.4t1 0.4l11.9 11.9 11.9-11.9q0.4-0.4 1-0.4t1 0.4l3.7 3.7q0.4 0.4 0.4 1t-0.4 1z"
                    />
                  </g>
                </svg>
              </div>
            </div>
            <div
              className="comment-content"
            >
              <div
                className="inline-edit-read-only inline-edit comment-body edit-input-unchanged"
              >
                Comments. Are. Cool.
              </div>
              <div
                className="action-bar"
              >
                <hr
                  className="action-bar-separator"
                />
                <span
                  className="action-bar-date"
                >
                   
                  &gt;1y
                   
                </span>
                <span
                  className="action-bar-author"
                >
                   by 
                  thingone
                   
                </span>
                <span
                  className="hidden"
                >
                   in 
                   
                </span>
                <span
                  className="action-bar-actions"
                >
                  [
                  <a
                    className="action-bar-action"
                    onClick={[Function]}
                  >
                     edit 
                  </a>
                  ]
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="post-wrapper"
    >
      <div
        className="post"
      >
        <div
          className="post-row"
        >
          <div
            className="rating"
          >
            <div
              id="upvote-button"
              onClick={[Function]}
            >
              <svg
                className="vote"
                fill="currentColor"
                height={14}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": "grey",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width={14}
              >
                <g>
                  <path
                    d="m37.6 29.7l-3.7 3.7q-0.5 0.4-1 0.4t-1-0.4l-11.9-11.9-11.8 11.9q-0.5 0.4-1.1 0.4t-1-0.4l-3.7-3.7q-0.4-0.4-0.4-1t0.4-1l16.6-16.6q0.4-0.4 1-0.4t1 0.4l16.6 16.6q0.4 0.4 0.4 1t-0.4 1z"
                  />
                </g>
              </svg>
            </div>
            <div
              className="rating-value"
              id="rating-value"
            >
              -5
            </div>
            <div
              id="downvote-button"
              onClick={[Function]}
            >
              <svg
                className="vote"
                fill="currentColor"
                height={14}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": "grey",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width={14}
              >
                <g>
                  <path
                    d="m37.6 18l-16.6 16.6q-0.4 0.4-1 0.4t-1-0.4l-16.6-16.6q-0.4-0.4-0.4-1t0.4-1l3.7-3.7q0.5-0.4 1-0.4t1 0.4l11.9 11.9 11.9-11.9q0.4-0.4 1-0.4t1 0.4l3.7 3.7q0.4 0.4 0.4 1t-0.4 1z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <div
            className="post-content"
          >
            <div
              className="inline-edit-read-only inline-edit post-title edit-input-unchanged"
            >
              Learn Redux in 10 minutes!
            </div>
            <div
              className="inline-edit-read-only inline-edit post-body edit-input-unchanged"
            >
              Just kidding. It takes more than 10 minutes to learn technology.
            </div>
            <div
              className="action-bar"
            >
              <hr
                className="action-bar-separator"
              />
              <span
                className="action-bar-date"
              >
                 
                &gt;1y
                 
              </span>
              <span
                className="action-bar-author"
              >
                 by 
                thingone
                 
              </span>
              <span
                className="action-bar-category"
              >
                 in 
                redux
                 
              </span>
              <span
                className="action-bar-actions"
              >
                [
                <a
                  className="action-bar-action"
                  onClick={[Function]}
                >
                   edit 
                </a>
                ]
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="posts-comments"
    />
  </div>
</div>
`;
